README.md

```diff
@@ -30,6 +30,54 @@ Any of the extensions listed below can be mixed into a class by `include`-ing `H
 
 ### Coercion
 
+Three systems are available for coercion: ActiveModel types, dry-types, and hashie's builtin types. The use of a specified coercion system is determined by exactly which Coercion module you include in your Hashie class. Including the default `Hashie::Extensions::Coercion` module will use Hashie's builtin type system.
+
+### Coercion using ActiveModel types
+
+To use ActiveModel's type coercion system, you must add version 5 of ActiveModel to your Gemfile. If you are using Rails, activemodel is included, but you must be using Rail's version 5 in order to use this. If you are on running a lower version of Rails, you will not be able to use ActiveModel's type coercion system due to Ruby's inability to use different Gem versions in the same application.
+
+```ruby
+gem 'activemodel', '~> 5.x'
+```
+
+Then include the `active_model` Coercion module into your Hashie class:
+
+```ruby
+class Tweet < Hash
+  include Hashie::Extensions::Coercion.active_model
```

Comment:
@michaelherold - I totally agree with everything you are saying about dynamic modules. What you a saying does not apply here though. When I said that `dry_types` returns a anonymous module, that anonymous module only has an `included` block defined and lives for a split second as it is being included into the base class. The modules that are mixed into the base class are static, so pry's `ls` method and the `ancestors` method are just as descriptive as before.

```ruby
mod = Hashie::Extensions::Coercion.dry_types
# => <Module: xyz>

mod.methods
#=> [:included]

class MyHash < Hashie::Hash
  include mod
end
#=> Hashie::Extensions::Coercion::ClassMethods, Hashie::Extensions::Coercion::InstanceMethods, and Hashie::Extensions::Coercion::DryTypes are all mixed into MyHash
```

Your second point is right. The primary reason for this is to remove duplicate code. Now that I explained that the first point is not an issue, perhaps code cuplication that is removed is more valuable? This is a tradeoff, and it is certainly to switch to simple way with code duplication. I should note that if we go this route, the `common include block` which mixes in the base Hashie::Coercion ClassMethods and InstanceMethods will need to duplicated 6 times throughout the code. Once for every coercion system type in both the `Coercion` and `Dash::Coercion` extension modules.

Please let me know you thoughts and instruct. :-)